src/lib/operators/index_scan.cpp

```diff
@@ -0,0 +1,154 @@
+#include "index_scan.hpp"
+
+#include <algorithm>
+
+#include "scheduler/abstract_task.hpp"
+#include "scheduler/current_scheduler.hpp"
+#include "scheduler/job_task.hpp"
+
+#include "storage/index/base_index.hpp"
+#include "storage/reference_column.hpp"
+
+#include "utils/assert.hpp"
+
+
+namespace opossum {
+
+IndexScan::IndexScan(std::shared_ptr<AbstractOperator> in, std::vector<ChunkID> chunk_ids,
+                     const ColumnIndexType index_type, std::vector<ColumnID> left_column_ids, const ScanType scan_type,
+                     std::vector<AllTypeVariant> right_values, std::vector<AllTypeVariant> right_values2)
```

Comment:
This is different from the TableScan interface, which takes `const std::optional<AllTypeVariant> right_value2`.

Reply:
Yes, maybe we should change it there too. `scan_type` indicates already if the value in `right_value2` is relevant or not. The additional optional just adds another boolean.

Reply:
What would we pass there instead of the optional? I am not a fan of ignored parameters.

Reply:
Just an `AllTypeVariant` was my idea. I mean, if `scan_type` is not `BETWEEN` we ignore the content of `right_value2` anyway, we don’t even check if the optional is set or not.

Reply:
Works for me.

Reply:
Or even better: `TableScan` does not need a second right value anymore, because it doesn’t support BETWEEN.

Reply:
Good point. Let's get rid of it.